app/src/main/java/framgia/vn/framgiacrb/ui/activity/LoginActivity.java

```diff
@@ -58,32 +58,28 @@ protected void onCreate(final Bundle savedInstanceState) {
         }
         mEditTextEmail = (EditText) findViewById(R.id.edit_email);
         mEditTextPassword = (EditText) findViewById(R.id.edit_password);
-        mButtonLogin = (Button) findViewById(R.id.btn_login);
-        mButtonLoginFb = (Button) findViewById(R.id.btn_facebook);
-        mButtonLoginGg = (Button) findViewById(R.id.btn_google);
         mProgressDialog = new ProgressDialog(LoginActivity.this);
-        mButtonLogin.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                boolean isValidEmail =
-                    ValidationLogin.isValidEmail(LoginActivity.this, mEditTextEmail);
-                boolean isValidPass =
-                    ValidationLogin.isValidatePassword(LoginActivity.this, mEditTextPassword);
-                if (!NetworkUtil.isInternetConnected(LoginActivity.this)) {
-                    Toast.makeText(LoginActivity.this, getString(R.string.msg_no_internet),
-                        Toast.LENGTH_SHORT).show();
-                } else {
-                    if (isValidEmail && isValidPass) {
-                        mProgressDialog.setMessage(Constant.LOADING);
-                        mProgressDialog.show();
-                        getDataFromInternet(user());
-                    } else {
-                        Toast.makeText(LoginActivity.this, getString(R.string.error_email_invalid),
-                            Toast.LENGTH_SHORT).show();
-                    }
-                }
+        findViewById(R.id.btn_login).setOnClickListener(this);
+    }
+
+    private void login() {
+        boolean isValidEmail =
+            ValidationLogin.isValidEmail(LoginActivity.this, mEditTextEmail);
+        boolean isValidPass =
+            ValidationLogin.isValidatePassword(LoginActivity.this, mEditTextPassword);
+        if (!NetworkUtil.isInternetConnected(LoginActivity.this)) {
+            Toast.makeText(LoginActivity.this, getString(R.string.msg_no_internet),
+                Toast.LENGTH_SHORT).show();
+        } else {
+            if (isValidEmail && isValidPass) {
+                mProgressDialog.setMessage(Constant.LOADING);
+                mProgressDialog.show();
+                getDataFromInternet(user());
+            } else {
+                Toast.makeText(LoginActivity.this, getString(R.string.error_email_invalid),
+                    Toast.LENGTH_SHORT).show();
```

Comment:
Cái này khi viết check nhiều sẽ khó nên viết như sau
if(!NetworkUtil.isInternetConnected(LoginActivity.this)){
todo something
return;
}
if(!isValidEmail ){
 todo something 
return;
}
if(!isValidPass){
todo something
return;
}
.... 
login